src/libraries/System.Threading.Tasks.Parallel/src/System/Threading/Tasks/Parallel.cs

```diff
@@ -1223,7 +966,7 @@ private static int ComputeTimeoutPoint(int timeoutLength)
             int numExpectedWorkers = (parallelOptions.EffectiveMaxConcurrencyLevel == -1) ?
                 Environment.ProcessorCount :
                 parallelOptions.EffectiveMaxConcurrencyLevel;
-            RangeManager rangeManager = new RangeManager(fromInclusive, toExclusive, 1, numExpectedWorkers);
+            RangeManager rangeManager = new RangeManager(long.CreateTruncating(fromInclusive), long.CreateTruncating(toExclusive), 1, numExpectedWorkers);
```

Comment:
Any reason why this wasn't made `RangeManager<TInt>` so it could exactly track the ranges?

Reply:
I didn't want to change any logic and have to review all the places or night possibly be relying on it actually being a long range. If it was already unifying int and long as long, I kept that without making it generic.